tests/repo_utils/test_check_copies.py

```diff
@@ -95,13 +95,139 @@ def forward(self, x):
+    def forward(self, c):
+        return 1
+
+    def existing_common(self, c):
+        return 4
+
+    def existing_diff_to_be_ignored(self, c):
+        return 9
+"""
+
+
+MOCK_DUMMY_ROBERTA_CODE_MATCH = """
+# Copied from transformers.models.dummy_bert_match.modeling_dummy_bert_match.BertDummyModel with BertDummy->RobertaBertDummy
+class RobertaBertDummyModel:
+    def __init__(self, a=1, b=2):
+        self.a = a
+        self.b = b
+
+    # ignore copied
+    def only_in_roberta_to_be_ignored(self, c):
+        return 3
+
+    # Copied from transformers.models.dummy_gpt2.modeling_dummy_gpt2.GPT2DummyModel.forward
+    def forward(self, c):
+        return 1
+
+    def existing_common(self, c):
+        return 4
+
+    # ignore copied
+    def existing_diff_to_be_ignored(self, c):
+        return 6
+"""
+
+
+MOCK_DUMMY_BERT_CODE_NOT_MATCH = """
+class BertDummyModel:
+    def __init__(self, a=1, b=2):
+        self.a = a
+        self.b = b
+
+    # Copied from transformers.models.dummy_gpt2.modeling_dummy_gpt2.GPT2DummyModel.forward
+    def forward(self, c):
+        return 1
+
+    def only_in_bert(self, c):
+        return 7
+
+    def existing_common(self, c):
+        return 4
+
+    def existing_diff_not_ignored(self, c):
+        return 8
+
+    def existing_diff_to_be_ignored(self, c):
+        return 9
+"""
+
+
+MOCK_DUMMY_ROBERTA_CODE_NOT_MATCH = """
+# Copied from transformers.models.dummy_bert_not_match.modeling_dummy_bert_not_match.BertDummyModel with BertDummy->RobertaBertDummy
+class RobertaBertDummyModel:
+    def __init__(self, a=1, b=2):
+        self.a = a
+        self.b = b
+
+    # ignore copied
+    def only_in_roberta_to_be_ignored(self, c):
+        return 3
+
+    # Copied from transformers.models.dummy_gpt2.modeling_dummy_gpt2.GPT2DummyModel.forward
+    def forward(self, c):
+        return 1
+
+    def only_in_roberta_not_ignored(self, c):
+        return 2
+
+    def existing_common(self, c):
+        return 4
+
+    def existing_diff_not_ignored(self, c):
+        return 5
+
+    # ignore copied
+    def existing_diff_to_be_ignored(self, c):
+        return 6
+"""
+
+
+EXPECTED_REPLACED_CODE = """
+# Copied from transformers.models.dummy_bert_not_match.modeling_dummy_bert_not_match.BertDummyModel with BertDummy->RobertaBertDummy
+class RobertaBertDummyModel:
+    def __init__(self, a=1, b=2):
+        self.a = a
+        self.b = b
+
+    # Copied from transformers.models.dummy_gpt2.modeling_dummy_gpt2.GPT2DummyModel.forward
+    def forward(self, c):
+        return 1
+
+    def only_in_bert(self, c):
+        return 7
+
+    def existing_common(self, c):
+        return 4
+
+    def existing_diff_not_ignored(self, c):
+        return 8
+
+    # ignore copied
+    def existing_diff_to_be_ignored(self, c):
+        return 6
+
+    # ignore copied
+    def only_in_roberta_to_be_ignored(self, c):
+        return 3
+"""
+
+
```

Comment:
Let's add the example where we want to ignore the class as well.

Reply:
As discussed offline last time, we can't use the on/off (for the individual lines). We can treat the whole first block (usually used for the class attributes) as a single block (as a block without any name), and in this case we don't need to add on/off.

But yes, I can add a test for this case.

Reply:
OK